contracts/PixelmonPartySquad/PsWeekManager.sol

```diff
@@ -0,0 +1,305 @@
+    /// @param tokenIds ERC721 Treasure token IDs
+    /// @param claimedToken Amount of token that has been claimed
+    /// @param contractType 1 for ERC1155, 2 for ERC721
+    /// @param treasureType Similar IDs for the treasure.Treasure ID is used
+    ///        to identify the treasure that claimed by winner and it's used to make
+    ///        sure the winner will get different set of prizes.
+    struct Treasure {
+        address collectionAddress;
+        uint256 tokenId;
+        uint256[] tokenIds;
+        uint256 claimedToken;
+        uint8 contractType;
+        uint8 treasureType;
+    }
+
+    /// @notice Struct object to store information about treasure that distributed within a week
+    /// @param treasureIndex Index of the treasure in the smart contract
+    /// @param totalSupply Total supply of the treasure within a week
+    struct TreasureDistribution {
+        uint8 treasureIndex;
+        uint16 totalSupply;
+    }
+
+    /// @notice Struct object to store week information
+    /// @param startTimeStamp Start timestamp of the week
+    /// @param ticketDrawTimeStamp ticket draw timestamp 
+    /// @param claimStartTimeStamp claiming start timestamp
+    /// @param endTimeStamp End timestamp of a week
+    /// @param remainingSupply The remaining treasure supply that hasn't been claimed during
+    ///        the week. This supply is the sum of every treasure supply excluding Special Treasures
+    /// @param treasureCount How many treasure option is available
+    /// @param specialTreasureCount How many Special Treasures are available in a week
+    /// @param specialTreasuresCount How many Special Treasures treasure that have not been claimed
+    /// @param specialTreasureWinners Winners of Special Treasures
+    /// @param specialTreasureWinnerMap Map that contains address of the Special Treasures winner.
+    ///        Map is used to easily validate whether the address is a winner rather than
+    ///        iterating every index in a list/array to find a winner
+    /// @param distributions Map of treasure that is distributed during the week
+    /// @param winners List of winner of the week
+    struct Week {
+        uint256 startTimeStamp;
+        uint256 ticketDrawTimeStamp;
+        uint256 claimStartTimeStamp;
+        uint256 endTimeStamp;
+        uint256 remainingSupply;
+        uint8 treasureCount;
+        uint8 specialTreasureCount;
+        uint8 availableSpecialTreasureCount;
+        address[] specialTreasureWinners;
+        mapping(address => bool) specialTreasureWinnerMap;
+        mapping(uint256 => TreasureDistribution) distributions;
+        mapping(address => Winner) winners;
+    }
+
+    /// @notice Struct object for week information
+    /// @dev This struct is only used as return type for getWeekInfo method
+    /// @param specialTreasureWinners Winner of Special Treasures
+    struct WeekData {
+        address[] specialTreasureWinners;
+        uint256[] randomNumbers;
+    }
```

Comment:
Would like to confirm, so the special treasure here would be E3 Serum since each wallet can get at max 5 serums, while the "normal" treasure is Relics, is that correct?

Reply:
special treasure is not fixed yet.